advanced_alchemy/base.py

```diff
@@ -126,8 +129,8 @@ class ModelProtocol(Protocol):
     """The base SQLAlchemy model protocol."""
 
     __table__: FromClause
-    __mapper__: Mapper
-    __name__: ClassVar[str]
+    __mapper__: Mapper[Any]
```

Comment:
In the SQLAlchemy version of this protocol, it is generic on a single type argument, which is used to type this mapper arg: https://github.com/sqlalchemy/sqlalchemy/blob/e69ca167fa0dcd33ef8c7dccbccf127c8665472e/lib/sqlalchemy/orm/decl_base.py#L90

I _think_ typing it as `Mapper[Any]` is OK for our purposes..